src/Microsoft.AspNetCore.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcBuilderExtensions.cs

```diff
@@ -67,6 +67,28 @@ public static IMvcBuilder AddViewComponentsAsServices(this IMvcBuilder builder)
         }
 
         /// <summary>
+        /// Registers <see cref="SessionStateTempDataProvider"/> as the default <see cref="ITempDataProvider"/>
+        /// in the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcBuilder"/>.</param>
+        /// <returns>The <see cref="IMvcBuilder"/>.</returns>
+        public static IMvcBuilder AddSessionStateTempDataProvider(this IMvcBuilder builder)
```

Comment:
The functional tests's BasicWebSite's  Startup file need to be updated to reflect this default temp data provider change.